backend/src/lib/repositories/dataset-repo.ts

```diff
@@ -51,6 +86,90 @@ class DatasetRepository extends BaseRepository {
       Item: DatasetFactory.toItem(dataset),
     });
   }
+
+  public async createDataset(metadata: any, data: any): Promise<Dataset> {
+    const jsonS3Key = this.getNewJsonS3Key();
+    const jsonKey = this.s3Prefix.concat(jsonS3Key);
+    await this.s3Service.putDataset(this.bucketName, jsonKey, data);
+    const dataset = DatasetFactory.createNew({
+      fileName: metadata.name,
+      createdBy: metadata.createdBy,
+      s3Key: { raw: "", json: jsonS3Key },
+      sourceType: SourceType.IngestApi,
+    });
+
+    let fileExists = await this.s3Service.objectExists(
+      this.bucketName,
+      jsonKey
+    );
+    if (!fileExists) {
+      console.error("JSON file not found for dataset=", dataset);
+      throw new Error("JSON file for dataset not found on S3");
+    }
+
+    this.dynamodb.put({
+      TableName: this.tableName,
+      Item: DatasetFactory.toItem(dataset),
+    });
+
+    return dataset;
+  }
+
+  public async updateDataset(id: string, metadata: any, data: any) {
+    const dataset = await this.getDatasetById(id);
+    const jsonS3Key = dataset.s3Key.json || this.getNewJsonS3Key();
+    const jsonKey = this.s3Prefix.concat(jsonS3Key);
+    await this.s3Service.putDataset(this.bucketName, jsonKey, data);
+
+    let fileExists = await this.s3Service.objectExists(
+      this.bucketName,
+      jsonKey
+    );
```

Comment:
Is this necessary? If you just uploaded the object on line 122 why do you need to check if it exists?